wordpress-mobile/WordPress-iOS
# Reader Post Details Comments: add functionality to `Follow Conversation` button

Ref: #17632 

This uses the `ReaderCommentsFollowPresenter` to display the `Follow Conversation` flow from the Comments snippet `Follow Conversation` button.

Notes:
- If following is toggled from the Comments view, the Comments snippet button is not updated on-the-fly yet.

To test:
- Enable the `followConversationPostDetails` feature flag.
- Go to Reader, and select any unsubscribed post.
- 🔍 Verify the "Follow Conversation" button is shown in the Comments header.
- Tap the Follow Conversation button.
- 🔍 Verify that the snackbar with the "Enable" action button is shown.
- Tap on the "Enable" button.
- 🔍 Verify that a feedback notice with an "Undo" action button is shown.
- Tap on the "Undo" action button.
- 🔍 Verify that a feedback notice is shown.
- 🔍 Verify that the Follow Conversation button has turned into a bell icon button.
- Go to Reader, and select any subscribed post.
- Tap on the bell icon button.
- 🔍 Verify that the bottom sheet is shown.
- Tap on the switch.
- Dismiss the bottom sheet.
- 🔍 Verify that a feedback notification is shown, and correctly represents the desired state.
- Tap the bell icon button again.
- Tap on the Unfollow Conversation button.
- 🔍 Verify that a feedback notification is shown.
- 🔍 Verify that the bell icon button has now turned back into "Follow Conversation".

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete and disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete and disabled.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete and disabled.

PR submission checklist:
